disco_aws_automation/disco_aws.py

```diff
@@ -715,13 +726,19 @@ def spinup(self, hostclass_dicts, stage=None, no_smoke=False, testing=False, cre
                 for (hostclass, termination_policies, hdict) in hostclass_iter]
 
             if metadata[0]:
+                for _hc in metadata:
+                    if _hc.get('spotinst'):
+                        self.elastigroup.wait_for_instance_id(_hc['group_name'])
                 self.smoketest(self.wait_for_autoscaling_instances(
                     [_hc for _hc in metadata if _hc["hostclass"] in flammable]))
 
     @staticmethod
     def _instance_count_lt_min_size(group, all_instances):
-        group_instances = [_i for _i in all_instances if _i.group_name == group.name]
-        return len(group_instances) < group.min_size
+        group_instances = [_i for _i in all_instances if _i['group_name'] == group['name']]
+        if group.get('id'):
```

Comment:
Add some comment explaining why we do this test and have those two option  or use class hierarchy to handle group class (AsgGroup and SpotGroup)
second option is probably best